rewriter/Struct.h

```diff
@@ -7,20 +7,41 @@ namespace sorbet::rewriter {
 /**
  * This class desugars things of the form
  *
- *   A = Struct.new(:foo, :bar)
+ *   A = Struct.new(:foo, :bar) do
+ *       # ... body ...
+ *   end
  *
  * into
  *
- *   class A < Struct
+ *   class A$1 < Struct
+ *       sig {returns(T.untyped)}
  *       def foo; end
+ *
+ *       sig {params(arg: T.untyped).returns(T.untyped)}
  *       def foo=(arg); arg; end
+ *
+ *       sig {returns(T.untyped)}
  *       def bar; end
+ *
+ *       sig {params(arg: T.untyped).returns(T.untyped)}
  *       def bar=(arg); arg; end
+ *
  *       sig {params(foo: BasicObject, bar: BasicObject).returns(A)}
- *       def self.new(foo=nil, bar=nil)
- *           T.cast(nil, A)
- *       end
+ *       def initialize(foo=nil, bar=nil); end
+ *
+ *       Elem = type_member {{fixed: T.untyped}}
```

Comment:
Potentially a breaking change: do we want to rename type_member to be something Struct specific like `StructElement`?  `Elem` is a reserved name for type members of `Enumerable`. However, a clash between Struct and Enumerable is pretty unlikely, but maybe we should give non-generic names to the type members and the type elements we use internally ? It will minimize a chance of potential collisions in general and also will be a good example to the users.

Reply:
This was already defined like this, and it has to be defined like this because `Struct` is `Enumerable`. That is, it's intentional that this uses the name `Elem`, because that's what `Enumerable` uses.

Reply:
https://github.com/sorbet/sorbet/blob/978f5f0c48fa18ea6c5d2835fd01247ac0c23f96/rbi/core/struct.rbi#L32-L35